### ISSUE TITLE
Reaching out from GitHub's The ReadME Project

### ISSUE THREAD

**Issue by mikethemelanson** [hollowaykeanho/hollowaykeanho#1]
Holloway! I am a senior editor with [The ReadME Project](https://github.com/readme/) here at GitHub and I just read your [introduction](https://github.com/community/maintainers/discussions/3#discussioncomment-5539025) in the /maintainers Discussion and it caught my eye for two reasons:

1. You mention being "very worried about the current geo-political effects on OSS distributions" and I'd like to hear more about what this means. 
2. I am researching a story about "the enduring promise of cross-platform development" and I see your Hestia project mention ideas around this point. I also see you mention interest in WASI/WASM. I'd be interested to hear more about this topic.

**Comment by hollowaykeanho** [comment]
Hi @mikethemelanson. Nice to meet you.


----

For item 1, quite a headache since its discovery back in year 2021 but things are a lot more smoother now. 
I wrote a paper about the matter here (https://zenodo.org/record/6815013). 

So you know most OSS are hosted on US infrastructure right? Then the US vs China trade war, US vs. Russia, etc. At the end of the day, GitHub must comply to US technogical embargo (I mentioned a few references in the paper related to GitHub cases). GitLab is also affected.

Most languages like Go and Rust encourage centralized sourcing (e.g. `go get github.com/...` and rust cargo) and if you're residing in countries not favoring US side (I'm Chinese ethics residing in a neutral Muslim country), you don't have a choice but to face the fear of losing the upstream access entirely someday in the future due to political shenanigan. The recent TikTok congressional grilling a Singaporean, also from a country from SEA, mistreated as a China citizen by some politicians is quite convincing that SEA can be a target of embargoment.

The solution is to move away from using these tools and go back to basic (a practice from C project back in the 2005). ZORALab's Hestia bans external sourcing and importing by default but to assimilate and rewrite all libraries. Both Rust and Go must use relative pathing in module files (`go.mod` & `cargo.toml`). The downside is the assimilation process takes time.

Similarly and also solved, my CI automations were migrated away from using vendor locked-in tools via ZORALab's Monteur Project. In short, GitHub Action and GitLab CI calls the ZORALab's Monteur instead of directly meddling with the repository's infrastructures. So, in case that day kicks in, I'm not affected as the localized system can easily pick up the changes.

I personally don't think US or EU citizens can empathize the mentioned fear and problem above. I mean, the problem won't affect them anyway.

----

For item 2, assimilating libraries for ZORALab's Hestia Project is an expensive effort. Since it is not a choice but to bulldoze the problem, might as well just solve a bunch more problems at the same time:

1. provides alternative language aside Javascript via WASM for frontend work. (Init PHASE 2).
2. Ensure high-reusability and make sure the codes are optimized to the point where microcontroller are also usable (Init PHASE 3).
3. Stabilize the community contributions and grow (Init PHASE 4).

I did some successful experimentation here: https://github.com/hollowaykeanho/ExperimentingGoWASM related WASM complete takes over. Now is to develop it in into ZORALab's Hestia.

Development progress is slow though . Been juggling with paying customers' projects for funding since March'23. Wish there is a channel to secure some angel funding to work on the project full-time.

**Comment by hollowaykeanho** [comment]
Hi [@mikethemelanson](https://github.com/mikethemelanson). Nice to meet you.

----

For item 1, quite a headache since its discovery back in year 2021 but things are a lot more smoother now.
I wrote a paper about the matter here (https://zenodo.org/record/6815013).

So you know most OSS are hosted on US infrastructure right? Then the US vs China trade war, US vs. Russia, etc. At the end of the day, GitHub must comply to US technogical embargo (I mentioned a few references in the paper related to GitHub cases). GitLab is also affected.

Most languages like Go and Rust encourage centralized sourcing (e.g. go get [github.com/.](http://github.com/).. and rust cargo) and if you're residing in countries not favoring US side (I'm Chinese ethics residing in a neutral Muslim country), you don't have a choice but to face the fear of losing the upstream access entirely someday in the future due to political shenanigan. The recent TikTok congressional grilling a Singaporean, also from a country from SEA, mistreated as a China citizen by some politicians is quite convincing that SEA can be a target of embargoment.

The solution is to move away from using these tools and go back to basic (a practice from C project back in the 2005). ZORALab's Hestia bans external sourcing and importing by default but to assimilate and rewrite all libraries. Both Rust and Go must use relative pathing in module files (go.mod & cargo.toml). The downside is the assimilation process takes time.

Similarly and also solved, my CI automations were migrated away from using vendor locked-in tools via ZORALab's Monteur Project. In short, GitHub Action and GitLab CI calls the ZORALab's Monteur instead of directly meddling with the repository's infrastructures. So, in case that day kicks in, I'm not affected as the localized system can easily pick up the changes.

I personally don't think US or EU citizens can empathize the mentioned fear and problem above. I mean, the problem won't affect them anyway.

---

For item 2, assimilating libraries for ZORALab's Hestia Project is an expensive effort. Since it is not a choice but to bulldoze the problem, might as well just solve a bunch more problems at the same time:

provides alternative language aside Javascript via WASM for frontend work. (Init PHASE 2).
Ensure high-reusability and make sure the codes are optimized to the point where microcontroller are also usable (Init PHASE 3).
Stabilize the community contributions and grow (Init PHASE 4).
I did some successful experimentation here: https://github.com/hollowaykeanho/ExperimentingGoWASM related WASM complete takes over. Now is to develop it in into ZORALab's Hestia.

Development progress is slow though . Been juggling with paying customers' projects for funding since March'23. Wish there is a channel to secure some angel funding to work on the project full-time.

----
